sunpy/map/tests/test_mapsequence.py

```diff
@@ -196,3 +198,23 @@ def test_norm_animator(hmi_test_map):
 def test_map_sequence_plot(aia171_test_map, hmi_test_map):
     seq = sunpy.map.Map([aia171_test_map, hmi_test_map], sequence=True)
     seq.plot()
+
+
+def test_save(aia171_test_map, hmi_test_map):
+    """Tests the map save function"""
+    seq = sunpy.map.Map([aia171_test_map, hmi_test_map], sequence=True)
+    afilename = tempfile.NamedTemporaryFile(suffix='fits').name
```

Comment:
This usage of `tempfile` is poor form.  You've created a temporary filename, but then you create two actual files with modified versions of that filename.  Not only have you opened yourself to the (admittedly unlikely) possibility of a filename collision, you aren't doing any cleanup of the files (and `tempfile` won't be able to clean them up automatically).

You should be using `tempfile.TemporaryDirectory()` and working within that space.

Reply:
I did use the same method used in `GenericMap` I shall make this change.

Reply:
I think we can update the code in GenericMap as well then.

Reply:
Should I use Pytest's `tmpdir` instead of `tempfile` here?

Reply:
Sure.

Reply:
> I think we can update the code in GenericMap as well then.

The code in the test for `GenericMap.save()` is fine because only one file is being created, and it's properly using the temporary filename provided by `tempfile`.  Thus, it was also be cleaned up automatically.